test/evergreen.yml

```diff
@@ -71,59 +71,64 @@ functions:
           fi
         fi
-
-pre:
-  - command: shell.exec
-    params:
-      script: |
-        rm -rf "wiredtiger"
-post:
-  - command: shell.exec
+  "make check directory":
+    command: shell.exec
     params:
-      working_dir: "wiredtiger"
+      working_dir: "wiredtiger/build_posix"
       script: |
         set -o errexit
         set -o verbose
-        tar cfz ../wiredtiger.tgz .
-  - command: s3.put
+
+        ${test_env_vars|} ${make_command|make} VERBOSE=1 check -C ${directory}  ${smp_command|} 2>&1
+  "upload artifact":
+    - command: archive.targz_pack
+      params:
+        target: "wiredtiger.tgz"
+        source_dir: "wiredtiger"
+        include:
+          - "./**"
+    - command: s3.put
+      params:
+        aws_secret: ${aws_secret}
+        aws_key: ${aws_key}
+        local_file: wiredtiger.tgz
+        bucket: build_external
+        permissions: public-read
+        content_type: application/tar
+        display_name: Artifacts
+        remote_file: wiredtiger/${build_variant}/${revision}/artifacts/${task_name}_${build_id}.tgz
+  "cleanup":
+    command: shell.exec
     params:
-      aws_secret: ${aws_secret}
-      aws_key: ${aws_key}
-      local_file: wiredtiger.tgz
-      bucket: build_external
-      permissions: public-read
-      content_type: application/tar
-      display_name: Artifacts
-      remote_file: wiredtiger/${build_variant}/${revision}/artifacts/${task_id}.tgz
+      script: |
+        rm -rf "wiredtiger"
+        rm -rf "wiredtiger.tgz"
+
+pre:
+  - func: "cleanup"
+post:
+  - func: "upload artifact"
+  - func: "cleanup"
 
 tasks:
 ## Base compile task on posix flavours
   - name: compile
     commands:
       - func: "get project"
       - func: "compile wiredtiger"
-      - command: archive.targz_pack
-        params:
-          target: "wiredtiger.tgz"
-          source_dir: "wiredtiger"
-          include:
-            - "./**"
-      - command: s3.put
-        params:
-          aws_secret: ${aws_secret}
-          aws_key: ${aws_key}
-          local_file: wiredtiger.tgz
-          bucket: build_external
-          permissions: public-read
-          content_type: application/tar
-          display_name: Artifacts_compile
-          remote_file: wiredtiger/${build_variant}/${revision}/artifacts/${build_id}.tgz
-      # Remove the artifacts here so the later post commands won't perform duplicated archiving.
-      - command: shell.exec
-        params:
-          script: |
-            rm -rf "wiredtiger"
-            rm -rf "wiredtiger.tgz"
+      - func: "upload artifact"
+      - func: "cleanup"
+
+  - name: compile-asan
+    commands:
+      - func: "get project"
+      - func: "compile wiredtiger"
+        vars:
+          smp_command: -j $(grep -c ^processor /proc/cpuinfo)
+          test_env_vars: PATH=/opt/mongodbtoolchain/v3/bin:$PATH LD_LIBRARY_PATH=$(pwd)/.libs top_srcdir=$(pwd)/.. top_builddir=$(pwd)
```

Comment:
Are they necessary considering the 2 variables are set as the same values as build variant level expansion?